### ISSUE TITLE
Feat: Food Cost Reports

### ISSUE THREAD

**Issue by ohdzdev** [ohdzdev/NutritionSystem#9]
**Is your feature request related to a problem? Please describe.**
* As a nutritionist / supervisor I need to see how much stuff is costing. We have all the #s in there to calculate cost of food (cost per gram). Now we need to create the reports for this stuff.

**Describe the solution you'd like**
* Under the new Reports sidebar entry create a component that has all the cost reports in either a printable output, or in an interactive way. Could be tabular or just layed out nicely.
* Use original application for details

**Describe alternatives you've considered**

**Additional context**

**Comment by ColinKrist** [comment]
**background information gathering**
Access 'Query' (essentially temp table) was created called `QryDietInG`. This compiled **every** diet's average food per day and on 'form view' only retrieved the relevant diet in the form.

**Form pulling info from QryDietInG:**
```sql
SELECT QryDietInG.DIETID, QryDietInG.FOOD, QryDietInG.SumOfAvgGPerDay, QryDietInG.UNIT
FROM QryDietInG
WHERE (((QryDietInG.DIETID)=[Forms]![DIET CARD ANALYSIS]![DIETID]));
```

**Forming QryDietInG**
Forming `QryDietInG` seems to be a bit more complicated and relied upon a `UNION ALL`  'sub-query called `UQryDietWeigtsInG`. 
**Access view of QryDietInG**
![Access view of SQL](https://user-images.githubusercontent.com/27971797/62822746-17578b80-bb4d-11e9-8865-44f6df4bf88e.png)
**QryDietInG's SQL**
```sql
SELECT UQryDietWeightsinG.FOOD, UQryDietWeightsinG.DIETID, Sum(UQryDietWeightsinG.AvgGPerDay) AS SumOfAvgGPerDay, Sum([COSTg]*[AvgGPerDay]) AS COST, UQryDietWeightsinG.UNIT
FROM ([UNIT TABLE] INNER JOIN UQryDietWeightsinG ON [UNIT TABLE].UNIT = UQryDietWeightsinG.UNIT) LEFT JOIN [FOOD TABLE] ON UQryDietWeightsinG.FOOD = [FOOD TABLE].FOOD
GROUP BY UQryDietWeightsinG.FOOD, UQryDietWeightsinG.DIETID, UQryDietWeightsinG.UNIT, [UNIT TABLE].UNITID
ORDER BY [UNIT TABLE].UNITID;
```

**UQryDietWeigtsInG**
'UQryDietWeigtsInG` seems to be the end of the line for going 'deeper'. It was made in SQL so there is no 'helpful' Access window
```sql
SELECT [FOOD TABLE].FOOD, Sum(([SU]+[M]+[T]+[W]+[R]+[F]+[S])*(-1/7)*[GROUP_AMOUNT]*[ConversionToG]) AS AvgGPerDay, "g" AS UNIT, [DIET SUB TABLE].DIETID
FROM [UNIT TABLE] INNER JOIN ([FOOD TABLE] INNER JOIN [DIET SUB TABLE] ON [FOOD TABLE].FOODID = [DIET SUB TABLE].FOODID) ON [UNIT TABLE].UNITID = [DIET SUB TABLE].UNITID
GROUP BY [FOOD TABLE].FOOD, "g", [DIET SUB TABLE].DIETID, [UNIT TABLE].UnitType
HAVING ((([UNIT TABLE].UnitType)="Weight"));

UNION ALL
SELECT [FOOD TABLE].FOOD, ([SU]+[M]+[T]+[W]+[R]+[F]+[S])*(-1/7)*[GROUP_AMOUNT] AS AvgGPerDay, [UNIT TABLE].UNIT, [DIET SUB TABLE].DIETID
FROM ([UNIT TABLE] INNER JOIN ([FOOD TABLE] INNER JOIN [DIET SUB TABLE] ON [FOOD TABLE].FOODID = [DIET SUB TABLE].FOODID) ON [UNIT TABLE].UNITID = [DIET SUB TABLE].UNITID) LEFT JOIN [WEIGHT TABLE] ON [FOOD TABLE].FOODID = [WEIGHT TABLE].FoodID
WHERE ((([UNIT TABLE].UnitType)="Volume") AND (([WEIGHT TABLE].[WtAmount]) Is Null) AND (([UNIT TABLE].UNITID)<>1));


UNION ALL SELECT [FOOD TABLE].FOOD, ([SU]+[M]+[T]+[W]+[R]+[F]+[S])*(-1/7)*[GROUP_AMOUNT]/[WtAmount]*[Gm_Wgt] AS AvgGPerDay, "g" AS Unit, [DIET SUB TABLE].DIETID
FROM ([UNIT TABLE] RIGHT JOIN ([FOOD TABLE] RIGHT JOIN [DIET SUB TABLE] ON [FOOD TABLE].FOODID = [DIET SUB TABLE].FOODID) ON [UNIT TABLE].UNITID = [DIET SUB TABLE].UNITID) LEFT JOIN [WEIGHT TABLE] ON ([DIET SUB TABLE].FOODID = [WEIGHT TABLE].FoodID) AND ([DIET SUB TABLE].UNITID = [WEIGHT TABLE].UnitIDNum)
WHERE (((([SU]+[M]+[T]+[W]+[R]+[F]+[S])*(-1/7)*[GROUP_AMOUNT]/[WtAmount]*[Gm_Wgt]) Is Not Null) AND (([UNIT TABLE].UnitType)="Volume"));
```

**Comment by ColinKrist** [comment]
the second select statement in UQryDietWeigtsInG has errors with grouping. Exploring solution

**Comment by ColinKrist** [comment]
resolved by #35
